include/p4mlir/Dialect/P4HIR/P4HIR_Types.td

```diff
@@ -88,6 +88,22 @@ def InfIntType : P4HIR_Type<"InfInt", "infint"> {
 
 }
 
+//===----------------------------------------------------------------------===//
+// VarBitsType: Bit-strings of dynamically-computed width
+// with a fixed maximum width
+//===----------------------------------------------------------------------===//
+
+def VarBitsType : P4HIR_Type<"VarBits", "varbit"> {
+  let summary = "bit-strings of dynamically-computed width with a fixed maximum width";
+  let description = [{
+    A variable-width bit-string type with a fixed maximum width.
+    The actual width is determined at runtime but cannot exceed the maximum.
+  }];
+
+  let parameters = (ins "unsigned":$maxWidth);
```

Comment:
Are we missing checks here?
 - varbit could be only in headers
 - it should be the last field (before validity)